// botframework-webchat/lib/adaptiveCards/Attachment/SignInCardAttachment.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _botframeworkWebchatComponent = require("botframework-webchat-component");

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireDefault(require("react"));

var _CommonCard = _interopRequireDefault(require("./CommonCard"));

var SignInCardAttachment = function SignInCardAttachment(_ref) {
  var adaptiveCardHostConfig = _ref.adaptiveCardHostConfig,
      adaptiveCards = _ref.adaptiveCards,
      attachment = _ref.attachment,
      styleSet = _ref.styleSet;
  return _react.default.createElement("div", {
    className: styleSet.animationCardAttachment
  }, _react.default.createElement(_CommonCard.default, {
    adaptiveCardHostConfig: adaptiveCardHostConfig,
    adaptiveCards: adaptiveCards,
    attachment: attachment
  }));
};

SignInCardAttachment.propTypes = {
  adaptiveCardHostConfig: _propTypes.default.any.isRequired,
  adaptiveCards: _propTypes.default.any.isRequired,
  attachment: _propTypes.default.any.isRequired,
  styleSet: _propTypes.default.shape({
    animationCardAttachment: _propTypes.default.any.isRequired,
    options: _propTypes.default.any.isRequired
  }).isRequired
};

var _default = (0, _botframeworkWebchatComponent.connectToWebChat)(function (_ref2) {
  var styleSet = _ref2.styleSet;
  return {
    styleSet: styleSet
  };
})(SignInCardAttachment);

exports.default = _default;
