// botframework-webchat-component/lib/Activity/Bubble.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _classnames = _interopRequireDefault(require("classnames"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireDefault(require("react"));

var _connectToWebChat = _interopRequireDefault(require("../connectToWebChat"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Bubble = function Bubble(_ref) {
  var ariaHidden = _ref['aria-hidden'],
      children = _ref.children,
      className = _ref.className,
      fromUser = _ref.fromUser,
      styleSet = _ref.styleSet;
  return _react.default.createElement("div", {
    "aria-hidden": ariaHidden,
    className: (0, _classnames.default)(styleSet.bubble + '', {
      'from-user': fromUser
    }, className + '')
  }, children);
};

Bubble.defaultProps = {
  'aria-hidden': true,
  children: undefined,
  className: '',
  fromUser: false
};
Bubble.propTypes = {
  'aria-hidden': _propTypes.default.bool,
  children: _propTypes.default.any,
  className: _propTypes.default.string,
  fromUser: _propTypes.default.bool,
  styleSet: _propTypes.default.shape({
    bubble: _propTypes.default.any.isRequired
  }).isRequired
};

var _default = (0, _connectToWebChat.default)(function (_ref2) {
  var styleSet = _ref2.styleSet;
  return {
    styleSet: styleSet
  };
})(Bubble);

exports.default = _default;
