// botframework-webchat-component/lib/Activity/CarouselLayout.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _reactFilm = require("react-film");

var _glamor = require("glamor");

var _classnames = _interopRequireDefault(require("classnames"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireDefault(require("react"));

var _Localize = require("../Localization/Localize");

var _connectToWebChat = _interopRequireDefault(require("../connectToWebChat"));

var _CarouselFilmStrip = _interopRequireDefault(require("./CarouselFilmStrip"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var ROOT_CSS = (0, _glamor.css)({
  overflow: 'hidden',
  position: 'relative'
});

var CarouselLayout = function CarouselLayout(_ref) {
  var activity = _ref.activity,
      children = _ref.children,
      language = _ref.language,
      styleSet = _ref.styleSet,
      timestampClassName = _ref.timestampClassName;
  var filmStyleSet = (0, _reactFilm.createBasicStyleSet)({
    cursor: null
  });
  return _react.default.createElement(_reactFilm.Composer, {
    numItems: _react.default.Children.count(children)
  }, _react.default.createElement(_reactFilm.Context.Consumer, null, function (_ref2) {
    var scrollBarWidth = _ref2.scrollBarWidth;
    return _react.default.createElement("div", {
      className: (0, _classnames.default)(ROOT_CSS + '', filmStyleSet.carousel + '')
    }, _react.default.createElement(_CarouselFilmStrip.default, {
      activity: activity,
      timestampClassName: timestampClassName
    }, children), scrollBarWidth !== '100%' && _react.default.createElement(_react.default.Fragment, null, _react.default.createElement(_reactFilm.Flipper, {
      "aria-label": (0, _Localize.localize)('Left', language),
      className: (0, _classnames.default)(styleSet.carouselFlipper + '', filmStyleSet.leftFlipper + ''),
      mode: "left"
    }, _react.default.createElement("div", {
      className: "button"
    }, '<')), _react.default.createElement(_reactFilm.Flipper, {
      "aria-label": (0, _Localize.localize)('Right', language),
      className: (0, _classnames.default)(styleSet.carouselFlipper + '', filmStyleSet.rightFlipper + ''),
      mode: "right"
    }, _react.default.createElement("div", {
      className: "button"
    }, '>'))));
  }));
};

CarouselLayout.defaultProps = {
  children: undefined,
  timestampClassName: ''
};
CarouselLayout.propTypes = {
  activity: _propTypes.default.any.isRequired,
  children: _propTypes.default.any,
  language: _propTypes.default.string.isRequired,
  styleSet: _propTypes.default.shape({
    carouselFlipper: _propTypes.default.any.isRequired
  }).isRequired,
  timestampClassName: _propTypes.default.string
};

var _default = (0, _connectToWebChat.default)(function (_ref3) {
  var language = _ref3.language,
      styleSet = _ref3.styleSet;
  return {
    language: language,
    styleSet: styleSet
  };
})(CarouselLayout);

exports.default = _default;
