// botframework-webchat-component/lib/BasicSendBox.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _botframeworkWebchatCore = require("botframework-webchat-core");

var _glamor = require("glamor");

var _classnames = _interopRequireDefault(require("classnames"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireDefault(require("react"));

var _ConnectivityStatus = _interopRequireDefault(require("./SendBox/ConnectivityStatus"));

var _connectToWebChat = _interopRequireDefault(require("./connectToWebChat"));

var _DictationInterims = _interopRequireDefault(require("./SendBox/DictationInterims"));

var _MicrophoneButton = _interopRequireDefault(require("./SendBox/MicrophoneButton"));

var _SendButton = _interopRequireDefault(require("./SendBox/SendButton"));

var _SuggestedActions = _interopRequireDefault(require("./SendBox/SuggestedActions"));

var _TextBox = _interopRequireDefault(require("./SendBox/TextBox"));

var _UploadButton = _interopRequireDefault(require("./SendBox/UploadButton"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _Constants$DictateSta = _botframeworkWebchatCore.Constants.DictateState,
    DICTATING = _Constants$DictateSta.DICTATING,
    STARTING = _Constants$DictateSta.STARTING;
var ROOT_CSS = (0, _glamor.css)({
  '& > .main': {
    display: 'flex'
  }
});
var DICTATION_INTERIMS_CSS = (0, _glamor.css)({
  flex: 10000
});
var MICROPHONE_BUTTON_CSS = (0, _glamor.css)({
  flex: 1
});
var TEXT_BOX_CSS = (0, _glamor.css)({
  flex: 10000
});

var BasicSendBox = function BasicSendBox(_ref) {
  var className = _ref.className,
      dictationStarted = _ref.dictationStarted,
      styleSet = _ref.styleSet,
      webSpeechPonyfill = _ref.webSpeechPonyfill;
  return _react.default.createElement("div", {
    className: (0, _classnames.default)(styleSet.sendBox + '', ROOT_CSS + '', className + ''),
    role: "form"
  }, _react.default.createElement(_ConnectivityStatus.default, null), _react.default.createElement(_SuggestedActions.default, null), _react.default.createElement("div", {
    className: "main"
  }, !styleSet.options.hideUploadButton && _react.default.createElement(_UploadButton.default, null), dictationStarted ? _react.default.createElement(_DictationInterims.default, {
    className: DICTATION_INTERIMS_CSS + ''
  }) : _react.default.createElement(_TextBox.default, {
    className: TEXT_BOX_CSS + ''
  }), _react.default.createElement("div", null, (webSpeechPonyfill || {}).SpeechRecognition ? _react.default.createElement(_MicrophoneButton.default, {
    className: MICROPHONE_BUTTON_CSS + ''
  }) : _react.default.createElement(_SendButton.default, null))));
};

BasicSendBox.defaultProps = {
  className: '',
  webSpeechPonyfill: undefined
};
BasicSendBox.propTypes = {
  className: _propTypes.default.string,
  dictationStarted: _propTypes.default.bool.isRequired,
  styleSet: _propTypes.default.shape({
    sendBox: _propTypes.default.any.isRequired
  }).isRequired,
  webSpeechPonyfill: _propTypes.default.shape({
    SpeechRecognition: _propTypes.default.any
  })
}; // TODO: [P3] We should consider exposing core/src/definitions and use it instead

function activityIsSpeakingOrQueuedToSpeak(_ref2) {
  var _ref2$channelData = _ref2.channelData;
  _ref2$channelData = _ref2$channelData === void 0 ? {} : _ref2$channelData;
  var speak = _ref2$channelData.speak;
  return !!speak;
}

var _default = (0, _connectToWebChat.default)(function (_ref3) {
  var activities = _ref3.activities,
      dictateState = _ref3.dictateState,
      styleSet = _ref3.styleSet,
      webSpeechPonyfill = _ref3.webSpeechPonyfill;
  return {
    dictationStarted: (dictateState === STARTING || dictateState === DICTATING) && !activities.filter(activityIsSpeakingOrQueuedToSpeak).length,
    styleSet: styleSet,
    webSpeechPonyfill: webSpeechPonyfill
  };
})(BasicSendBox);

exports.default = _default;
