// botframework-webchat-component/lib/ErrorBox.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireDefault(require("react"));

var _connectToWebChat = _interopRequireDefault(require("./connectToWebChat"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var ErrorBox = function ErrorBox(_ref) {
  var children = _ref.children,
      message = _ref.message,
      styleSet = _ref.styleSet;
  return _react.default.createElement("div", {
    className: styleSet.errorBox
  }, _react.default.createElement("div", null, message), _react.default.createElement("div", null, children));
};

ErrorBox.defaultProps = {
  children: undefined,
  message: ''
};
ErrorBox.propTypes = {
  children: _propTypes.default.any,
  message: _propTypes.default.string,
  styleSet: _propTypes.default.shape({
    errorBox: _propTypes.default.any.isRequired
  }).isRequired
};

var _default = (0, _connectToWebChat.default)(function (_ref2) {
  var styleSet = _ref2.styleSet;
  return {
    styleSet: styleSet
  };
})(ErrorBox);

exports.default = _default;
