// botframework-webchat-component/lib/Attachment/AudioContent.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireDefault(require("react"));

var _connectToWebChat = _interopRequireDefault(require("../connectToWebChat"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var AudioContent = function AudioContent(_ref) {
  var autoPlay = _ref.autoPlay,
      loop = _ref.loop,
      poster = _ref.poster,
      src = _ref.src,
      styleSet = _ref.styleSet;
  return _react.default.createElement("audio", {
    autoPlay: autoPlay,
    className: styleSet.audioContent,
    controls: true,
    loop: loop,
    poster: poster,
    src: src
  });
};

AudioContent.defaultProps = {
  autoPlay: false,
  loop: false,
  poster: ''
};
AudioContent.propTypes = {
  autoPlay: _propTypes.default.bool,
  loop: _propTypes.default.bool,
  poster: _propTypes.default.string,
  src: _propTypes.default.string.isRequired,
  styleSet: _propTypes.default.shape({
    audioContent: _propTypes.default.any.isRequired
  }).isRequired
};

var _default = (0, _connectToWebChat.default)(function (_ref2) {
  var styleSet = _ref2.styleSet;
  return {
    styleSet: styleSet
  };
})(AudioContent);

exports.default = _default;
