// botframework-webchat-component/lib/Attachment/AudioAttachment.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireDefault(require("react"));

var _AudioContent = _interopRequireDefault(require("./AudioContent"));

var _connectToWebChat = _interopRequireDefault(require("../connectToWebChat"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var AudioAttachment = function AudioAttachment(_ref) {
  var attachment = _ref.attachment,
      styleSet = _ref.styleSet;
  return _react.default.createElement("div", {
    className: styleSet.audioAttachment
  }, _react.default.createElement(_AudioContent.default, {
    alt: attachment.name,
    src: attachment.contentUrl
  }));
};

AudioAttachment.propTypes = {
  attachment: _propTypes.default.shape({
    contentUrl: _propTypes.default.string.isRequired,
    name: _propTypes.default.string
  }).isRequired,
  styleSet: _propTypes.default.shape({
    audioAttachment: _propTypes.default.any.isRequired
  }).isRequired
};

var _default = (0, _connectToWebChat.default)(function (_ref2) {
  var styleSet = _ref2.styleSet;
  return {
    styleSet: styleSet
  };
})(AudioAttachment);

exports.default = _default;
