// botframework-webchat-component/lib/Attachment/DownloadAttachment.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _bytes = require("bytes");

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireDefault(require("react"));

var _connectToWebChat = _interopRequireDefault(require("../connectToWebChat"));

var _DownloadIcon = _interopRequireDefault(require("./Assets/DownloadIcon"));

var _Localize = require("../Localization/Localize");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var DownloadAttachment = function DownloadAttachment(_ref) {
  var _ref$activity = _ref.activity;
  _ref$activity = _ref$activity === void 0 ? {} : _ref$activity;
  var _ref$activity$attachm = _ref$activity.attachments,
      attachments = _ref$activity$attachm === void 0 ? [] : _ref$activity$attachm,
      _ref$activity$channel = _ref$activity.channelData;
  _ref$activity$channel = _ref$activity$channel === void 0 ? {} : _ref$activity$channel;
  var _ref$activity$channel2 = _ref$activity$channel.attachmentSizes,
      attachmentSizes = _ref$activity$channel2 === void 0 ? [] : _ref$activity$channel2,
      attachment = _ref.attachment,
      language = _ref.language,
      styleSet = _ref.styleSet;
  var attachmentIndex = attachments.indexOf(attachment);
  var label = (0, _Localize.localize)('Download file', language);
  var size = attachmentSizes[attachmentIndex];
  return _react.default.createElement("div", {
    className: styleSet.downloadAttachment
  }, _react.default.createElement("a", {
    href: attachment.contentUrl,
    rel: "noopener noreferrer",
    target: "_blank"
  }, _react.default.createElement("div", {
    className: "details"
  }, _react.default.createElement("div", {
    className: "name"
  }, attachment.name), typeof size === 'number' && _react.default.createElement("div", {
    className: "size"
  }, (0, _bytes.format)(size))), _react.default.createElement(_DownloadIcon.default, {
    className: "icon",
    label: label,
    size: 1.5
  })));
};

DownloadAttachment.propTypes = {
  activity: _propTypes.default.shape({
    attachment: _propTypes.default.array,
    channelData: _propTypes.default.shape({
      attachmentSizes: _propTypes.default.arrayOf(_propTypes.default.number)
    })
  }).isRequired,
  attachment: _propTypes.default.shape({
    contentUrl: _propTypes.default.string.isRequired,
    name: _propTypes.default.string.isRequired
  }).isRequired,
  language: _propTypes.default.string.isRequired,
  styleSet: _propTypes.default.shape({
    downloadAttachment: _propTypes.default.any.isRequired
  }).isRequired
};

var _default = (0, _connectToWebChat.default)(function (_ref2) {
  var language = _ref2.language,
      styleSet = _ref2.styleSet;
  return {
    language: language,
    styleSet: styleSet
  };
})(DownloadAttachment);

exports.default = _default;
