// botframework-webchat-component/lib/Attachment/ImageContent.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireDefault(require("react"));

var _connectToWebChat = _interopRequireDefault(require("../connectToWebChat"));

var _CroppedImage = _interopRequireDefault(require("../Utils/CroppedImage"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var ImageContent = function ImageContent(_ref) {
  var alt = _ref.alt,
      src = _ref.src,
      styleSet = _ref.styleSet;
  return _react.default.createElement(_CroppedImage.default, {
    alt: alt,
    height: styleSet.options.bubbleImageHeight,
    src: src,
    width: "100%"
  });
};

ImageContent.defaultProps = {
  alt: ''
};
ImageContent.propTypes = {
  alt: _propTypes.default.string,
  src: _propTypes.default.string.isRequired,
  styleSet: _propTypes.default.shape({
    options: _propTypes.default.shape({
      bubbleImageHeight: _propTypes.default.number.isRequired
    }).isRequired
  }).isRequired
};

var _default = (0, _connectToWebChat.default)(function (_ref2) {
  var styleSet = _ref2.styleSet;
  return {
    styleSet: styleSet
  };
})(ImageContent);

exports.default = _default;
