// botframework-webchat-component/lib/Attachment/TextContent.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _classnames = _interopRequireDefault(require("classnames"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireDefault(require("react"));

var _connectToWebChat = _interopRequireDefault(require("../connectToWebChat"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// The content rendered here is sanitized.

/* eslint react/no-danger: "off" */

/* eslint react/no-array-index-key: "off" */
var TextContent = function TextContent(_ref) {
  var contentType = _ref.contentType,
      renderMarkdown = _ref.renderMarkdown,
      styleSet = _ref.styleSet,
      text = _ref.text;
  return contentType === 'text/markdown' && renderMarkdown ? _react.default.createElement("div", {
    className: (0, _classnames.default)('markdown', styleSet.textContent + ''),
    dangerouslySetInnerHTML: {
      __html: renderMarkdown(text || '')
    }
  }) : (text || '').split('\n').map(function (line, index) {
    return _react.default.createElement("p", {
      className: (0, _classnames.default)('plain', styleSet.textContent + ''),
      key: index
    }, line.trim());
  });
};

TextContent.defaultProps = {
  contentType: 'text/markdown',
  renderMarkdown: function renderMarkdown(text) {
    return text;
  }
};
TextContent.propTypes = {
  contentType: _propTypes.default.string,
  renderMarkdown: _propTypes.default.func,
  styleSet: _propTypes.default.shape({
    textContent: _propTypes.default.any.isRequired
  }).isRequired,
  text: _propTypes.default.string.isRequired
};

var _default = (0, _connectToWebChat.default)(function (_ref2) {
  var renderMarkdown = _ref2.renderMarkdown,
      styleSet = _ref2.styleSet;
  return {
    renderMarkdown: renderMarkdown,
    styleSet: styleSet
  };
})(TextContent);

exports.default = _default;
