// botframework-webchat-component/lib/Localization/nl-NL.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

/* eslint no-magic-numbers: ["error", { "ignore": [1, 5, 24, 48, 60000, 3600000] }] */
function xMinutesAgo(dateStr) {
  var date = new Date(dateStr);
  var dateTime = date.getTime();

  if (isNaN(dateTime)) {
    return dateStr;
  }

  var now = Date.now();
  var deltaInMs = now - dateTime;
  var deltaInMinutes = Math.floor(deltaInMs / 60000);
  var deltaInHours = Math.floor(deltaInMs / 3600000);

  if (deltaInMinutes < 1) {
    return 'Zojuist';
  } else if (deltaInMinutes === 1) {
    return 'Een minuut geleden';
  } else if (deltaInHours < 1) {
    return "".concat(deltaInMinutes, " minuten geleden");
  } else if (deltaInHours === 1) {
    return "Een uur geleden";
  } else if (deltaInHours < 5) {
    return "".concat(deltaInHours, " uur geleden");
  } else if (deltaInHours <= 24) {
    return "Vandaag";
  } else if (deltaInHours <= 48) {
    return "Gisteren";
  } else if (window.Intl) {
    return new Intl.DateTimeFormat('nl-NL').format(date);
  }

  return date.toLocaleString('nl-NL', {
    day: '2-digit',
    hour: '2-digit',
    hour12: false,
    minute: '2-digit',
    month: '2-digit',
    year: 'numeric'
  });
}

function botSaidSomething(avatarInitials, text) {
  return "Bot ".concat(avatarInitials, " zei; ").concat(text);
}

function userSaidSomething(avatarInitials, text) {
  return "Gebruiker ".concat(avatarInitials, " zei; ").concat(text);
}

var _default = {
  FAILED_CONNECTION_NOTIFICATION: 'Verbinding maken niet mogelijk.',
  SEND_FAILED_KEY: 'Versturen mislukt, {Retry}.',
  SLOW_CONNECTION_NOTIFICATION: 'Verbinding maken duurt langer dan normaal…',
  'Bot said something': botSaidSomething,
  Chat: 'Chat',
  'Download file': 'Bestand downloaden',
  'Listening…': 'Aan het luisteren…',
  'Microphone off': 'Microfoon uit',
  'Microphone on': 'Microfoon aan',
  Retry: 'probeer opnieuw',
  Send: 'Verstuur',
  Sending: 'versturen',
  Speak: 'Spreek',
  'Starting…': 'Starten…',
  Tax: 'BTW',
  Total: 'Totaal',
  'Type your message': 'Typ je bericht',
  'Upload file': 'Bestand uploaden',
  'User said something': userSaidSomething,
  VAT: 'VAT',
  'X minutes ago': xMinutesAgo
};
exports.default = _default;
