// botframework-webchat-component/lib/Localization/ru-RU.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

/* eslint no-magic-numbers: ["error", { "ignore": [1, 5, 24, 48, 72, 60000, 3600000] }] */
function xMinutesAgo(dateStr) {
  var date = new Date(dateStr);
  var dateTime = date.getTime();

  if (isNaN(dateTime)) {
    return dateStr;
  }

  var now = Date.now();
  var deltaInMs = now - dateTime;
  var deltaInMinutes = Math.floor(deltaInMs / 60000);
  var deltaInHours = Math.floor(deltaInMs / 3600000);

  if (deltaInMinutes < 1) {
    return 'Сейчас';
  } else if (deltaInMinutes === 1) {
    return 'Минуту назад';
  } else if (deltaInHours < 1 && deltaInMinutes < 5) {
    return "".concat(deltaInMinutes, " \u043C\u0438\u043D\u0443\u0442\u044B \u043D\u0430\u0437\u0430\u0434");
  } else if (deltaInHours < 1 && deltaInMinutes >= 5) {
    return "".concat(deltaInMinutes, " \u043C\u0438\u043D\u0443\u0442 \u043D\u0430\u0437\u0430\u0434");
  } else if (deltaInHours === 1) {
    return 'Час назад';
  } else if (deltaInHours < 5) {
    return "".concat(deltaInHours, " \u0447\u0430\u0441\u0430 \u043D\u0430\u0437\u0430\u0434");
  } else if (deltaInHours <= 24) {
    return 'Сегодня';
  } else if (deltaInHours <= 48) {
    return 'Вчера';
  } else if (deltaInHours <= 72) {
    return 'Позавчера';
  } else if (window.Intl) {
    return new Intl.DateTimeFormat('ru-RU').format(date);
  }

  return date.toLocaleString('ru-RU', {
    day: '2-digit',
    hour: '2-digit',
    minute: '2-digit',
    month: '2-digit',
    year: 'numeric'
  });
}

var _default = {
  FAILED_CONNECTION_NOTIFICATION: 'Ошибка подключения.',
  // Do not localize {Retry}; it is a placeholder for "Retry". English translation should be, "Send failed. Retry."
  SEND_FAILED_KEY: "\u041D\u0435 \u0443\u0434\u0430\u043B\u043E\u0441\u044C \u043E\u0442\u043F\u0440\u0430\u0432\u0438\u0442\u044C, {Retry}.",
  SLOW_CONNECTION_NOTIFICATION: 'Требуется больше времени, чем обычно.',
  'Adaptive Card parse error': 'Ошибка парсинга адаптивной карты',
  'Adaptive Card render error': 'Ошибка отображения адаптивной карты',
  Chat: 'Чат',
  'Download file': 'Скачать файл',
  'Microphone off': 'Микрофон влючен',
  'Microphone on': 'Микрофон выключен',
  'Listening…': 'Прослушивание…',
  Retry: 'повторить',
  Send: 'Отправить',
  Sending: 'Отправка',
  Speak: 'Говорить',
  'Starting…': 'Запуск…',
  Tax: 'Налог',
  Total: 'Итого',
  'Type your message': 'Введите ваше сообщение',
  'Upload file': 'Загрузить файл',
  VAT: 'НДС',
  'New messages': 'Новые сообщения',
  'X minutes ago': xMinutesAgo
};
exports.default = _default;
