// botframework-webchat-component/lib/Localization/tr-TR.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

/* eslint no-magic-numbers: ["error", { "ignore": [1, 5, 24, 48, 60000, 3600000] }] */
function xMinutesAgo(dateStr) {
  var date = new Date(dateStr);
  var dateTime = date.getTime();

  if (isNaN(dateTime)) {
    return dateStr;
  }

  var now = Date.now();
  var deltaInMs = now - dateTime;
  var deltaInMinutes = Math.floor(deltaInMs / 60000);
  var deltaInHours = Math.floor(deltaInMs / 3600000);

  if (deltaInMinutes < 1) {
    return 'Şimdi';
  } else if (deltaInMinutes === 1) {
    return 'Bir dakika önce';
  } else if (deltaInHours < 1) {
    return "".concat(deltaInMinutes, " dakika \xF6nce");
  } else if (deltaInHours === 1) {
    return "Bir saat \xF6nce";
  } else if (deltaInHours < 5) {
    return "".concat(deltaInHours, " saat \xF6nce");
  } else if (deltaInHours <= 24) {
    return "Bug\xFCn";
  } else if (deltaInHours <= 48) {
    return "D\xFCn";
  } else if (window.Intl) {
    return new Intl.DateTimeFormat('tr-TR').format(date);
  }

  return date.toLocaleString('tr-TR', {
    day: '2-digit',
    hour: '2-digit',
    hour12: false,
    minute: '2-digit',
    month: '2-digit',
    year: 'numeric'
  });
}

var _default = {
  FAILED_CONNECTION_NOTIFICATION: 'Bağlanamadı.',
  INITIAL_CONNECTION_NOTIFICATION: 'Bağlanıyor…',
  INTERRUPTED_CONNECTION_NOTIFICATION: 'Ağ kesintisi meydana geldi. Yeniden bağlanıyor...',
  // Do not localize {Retry}; it is a placeholder for "Retry". English translation should be, "Send failed. Retry."
  SEND_FAILED_KEY: "g\xF6nderilemedi, {Retry}.",
  SLOW_CONNECTION_NOTIFICATION: 'Bağlantı hızı çok düşük.',
  Chat: 'Sohbet',
  'Download file': 'Dosyayı indir',
  'Microphone off': 'Mikrofon kapalı',
  'Microphone on': 'Mikrofon açık',
  'X minutes ago': xMinutesAgo,
  'Listening…': 'Dinliyor…',
  Retry: 'yeniden deneyin',
  Send: 'Gönder',
  Sending: 'gönderiliyor',
  // 'Speak': '',
  'Starting…': 'Başlıyor…',
  Tax: 'Vergi',
  Total: 'Toplam',
  'Type your message': 'İletinizi yazın',
  // 'Upload file': '',
  VAT: 'KDV'
};
exports.default = _default;
