// botframework-webchat-component/lib/SendBox/DictationInterims.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.connectDictationInterims = exports.default = void 0;

var _glamor = require("glamor");

var _classnames = _interopRequireDefault(require("classnames"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireDefault(require("react"));

var _botframeworkWebchatCore = require("botframework-webchat-core");

var _connectToWebChat = _interopRequireDefault(require("../connectToWebChat"));

var _Localize = _interopRequireDefault(require("../Localization/Localize"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/* eslint react/no-array-index-key: "off" */
var _Constants$DictateSta = _botframeworkWebchatCore.Constants.DictateState,
    DICTATING = _Constants$DictateSta.DICTATING,
    STARTING = _Constants$DictateSta.STARTING,
    STOPPING = _Constants$DictateSta.STOPPING;
var ROOT_CSS = (0, _glamor.css)({
  alignItems: 'center',
  display: 'flex'
});

var connectDictationInterims = function connectDictationInterims() {
  for (var _len = arguments.length, selectors = new Array(_len), _key = 0; _key < _len; _key++) {
    selectors[_key] = arguments[_key];
  }

  return _connectToWebChat.default.apply(void 0, [function (_ref) {
    var dictateInterims = _ref.dictateInterims,
        dictateState = _ref.dictateState,
        language = _ref.language;
    return {
      dictateInterims: dictateInterims,
      dictateState: dictateState,
      language: language
    };
  }].concat(selectors));
};

exports.connectDictationInterims = connectDictationInterims;

var DictationInterims = function DictationInterims(_ref2) {
  var className = _ref2.className,
      _ref2$dictateInterims = _ref2.dictateInterims,
      dictateInterims = _ref2$dictateInterims === void 0 ? [] : _ref2$dictateInterims,
      dictateState = _ref2.dictateState,
      styleSet = _ref2.styleSet;
  return dictateState === STARTING || dictateState === STOPPING ? _react.default.createElement("p", {
    className: (0, _classnames.default)(styleSet.dictationInterims + '', ROOT_CSS + '', className + '', 'status')
  }, dictateState === STARTING && _react.default.createElement(_Localize.default, {
    text: "Starting\u2026"
  })) : dictateState === DICTATING && (dictateInterims.length ? _react.default.createElement("p", {
    className: (0, _classnames.default)(styleSet.dictationInterims + '', ROOT_CSS + '', className + '', 'dictating')
  }, dictateInterims.map(function (interim, index) {
    return _react.default.createElement("span", {
      key: index
    }, interim, "\xA0");
  })) : _react.default.createElement("p", {
    className: (0, _classnames.default)(styleSet.dictationInterims + '', ROOT_CSS + '', className + '', 'status')
  }, _react.default.createElement(_Localize.default, {
    text: "Listening\u2026"
  })));
};

DictationInterims.defaultProps = {
  className: ''
};
DictationInterims.propTypes = {
  className: _propTypes.default.string,
  dictateInterims: _propTypes.default.arrayOf(_propTypes.default.string).isRequired,
  dictateState: _propTypes.default.number.isRequired,
  styleSet: _propTypes.default.shape({
    dictationInterims: _propTypes.default.any.isRequired
  }).isRequired
}; // TODO: [P3] After speech started, when clicking on the transcript, it should
//       stop the dictation and allow the user to type-correct the transcript

var _default = connectDictationInterims(function (_ref3) {
  var styleSet = _ref3.styleSet;
  return {
    styleSet: styleSet
  };
})(DictationInterims);

exports.default = _default;
