// botframework-webchat-component/lib/SendBox/MicrophoneButton.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.connectMicrophoneButton = exports.default = void 0;

var _botframeworkWebchatCore = require("botframework-webchat-core");

var _glamor = require("glamor");

var _classnames = _interopRequireDefault(require("classnames"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireDefault(require("react"));

var _Localize = require("../Localization/Localize");

var _connectToWebChat = _interopRequireDefault(require("../connectToWebChat"));

var _IconButton = _interopRequireDefault(require("./IconButton"));

var _MicrophoneIcon = _interopRequireDefault(require("./Assets/MicrophoneIcon"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// This is required for aria-controls.

/* eslint react/forbid-dom-props: "off" */
var DictateState = _botframeworkWebchatCore.Constants.DictateState;
var ROOT_CSS = (0, _glamor.css)({
  display: 'flex',
  height: '100%',
  '& > .sr-only': {
    color: 'transparent',
    height: 1,
    left: -10000,
    overflow: 'hidden',
    position: 'absolute',
    top: 'auto',
    width: 1
  }
});

var connectMicrophoneButton = function connectMicrophoneButton() {
  for (var _len = arguments.length, selectors = new Array(_len), _key = 0; _key < _len; _key++) {
    selectors[_key] = arguments[_key];
  }

  return _connectToWebChat.default.apply(void 0, [function (_ref) {
    var disabled = _ref.disabled,
        dictateInterims = _ref.dictateInterims,
        dictateState = _ref.dictateState,
        language = _ref.language,
        setSendBox = _ref.setSendBox,
        startDictate = _ref.startDictate,
        stopDictate = _ref.stopDictate;
    return {
      click: function click() {
        if (dictateState === DictateState.STARTING || dictateState === DictateState.DICTATING) {
          stopDictate();
          setSendBox(dictateInterims.join(' '));
        } else {
          startDictate();
        }
      },
      dictating: dictateState === DictateState.DICTATING,
      disabled: disabled || dictateState === DictateState.STARTING || dictateState === DictateState.STOPPING,
      language: language
    };
  }].concat(selectors));
};

exports.connectMicrophoneButton = connectMicrophoneButton;

var MicrophoneButton = function MicrophoneButton(_ref2) {
  var className = _ref2.className,
      click = _ref2.click,
      dictating = _ref2.dictating,
      disabled = _ref2.disabled,
      language = _ref2.language,
      styleSet = _ref2.styleSet;
  return _react.default.createElement("div", {
    "aria-controls": "webchatSendBoxMicrophoneButton",
    className: (0, _classnames.default)(styleSet.microphoneButton + '', ROOT_CSS + '', className + '', {
      dictating: dictating
    })
  }, _react.default.createElement(_IconButton.default, {
    alt: (0, _Localize.localize)('Speak', language),
    disabled: disabled,
    onClick: click
  }, _react.default.createElement(_MicrophoneIcon.default, null)), _react.default.createElement("div", {
    "aria-live": "polite",
    className: "sr-only",
    id: "webchatSendBoxMicrophoneButton",
    role: "status"
  }, (0, _Localize.localize)(dictating ? 'Microphone on' : 'Microphone off', language)));
};

MicrophoneButton.defaultProps = {
  className: '',
  dictating: false,
  disabled: false
};
MicrophoneButton.propTypes = {
  className: _propTypes.default.string,
  click: _propTypes.default.func.isRequired,
  dictating: _propTypes.default.bool,
  disabled: _propTypes.default.bool,
  language: _propTypes.default.string.isRequired,
  styleSet: _propTypes.default.shape({
    microphoneButton: _propTypes.default.any.isRequired
  }).isRequired
};

var _default = connectMicrophoneButton(function (_ref3) {
  var styleSet = _ref3.styleSet;
  return {
    styleSet: styleSet
  };
})(MicrophoneButton);

exports.default = _default;
